torch/distributed/fsdp/_common_utils.py

```diff
@@ -278,14 +278,15 @@ def f(module: torch.nn.Module, prefix: str, *args, **kwargs):
                     for fqn in filter_fqns:
                         if fqn.startswith(new_prefix):
                             break
-                    else:
-                        # TODO: Remove this hack once DMP + FSDP is not supported.
-                        warnings.warn(
-                            "An unexpected prefix is detected. "
-                            "This case should only happen when using "
-                            "DistributedModelParallel with FullyShardedDataParallel."
-                        )
-                        new_prefix = prefix
+                        else:
```

Comment:
The original `else` is intended to be matched with `for` so that the warning message will only be printed when the prefix do not match any of the `filter_fqns`. If we match the `else` to `if`, the behavior is incorrect.